frontend/src/employee-frontend/components/child-information/placements/NewServiceNeeds.tsx

```diff
@@ -2,34 +2,23 @@
 //
 // SPDX-License-Identifier: LGPL-2.1-or-later
 
-import { Table, Tbody, Td, Th, Thead, Tr } from 'lib-components/layout/Table'
-import { H4 } from 'lib-components/typography'
-import React, { useContext, useState } from 'react'
-import { NewServiceNeed, Placement } from '../../../types/child'
+import React, { useContext, useMemo, useState } from 'react'
+import styled from 'styled-components'
 import _ from 'lodash'
+import { faPlus, faQuestion } from 'lib-icons'
+import LocalDate from 'lib-common/local-date'
+import InlineButton from 'lib-components/atoms/buttons/InlineButton'
+import { Table, Tbody, Th, Thead, Tr } from 'lib-components/layout/Table'
+import InfoModal from 'lib-components/molecules/modals/InfoModal'
+import { H4 } from 'lib-components/typography'
 import { useTranslation } from '../../../state/i18n'
-import Toolbar from '../../common/Toolbar'
-import styled from 'styled-components'
-import InlineButton from '../../../../lib-components/atoms/buttons/InlineButton'
-import { faPlus } from 'lib-icons'
-import LocalDate from '../../../../lib-common/local-date'
-import { UUID } from '../../../types'
-import { FixedSpaceRow } from 'lib-components/layout/flex-helpers'
-import { DatePickerDeprecated } from '../../../../lib-components/molecules/DatePickerDeprecated'
-import Checkbox from '../../../../lib-components/atoms/form/Checkbox'
-import SimpleSelect from '../../../../lib-components/atoms/form/SimpleSelect'
+import { NewServiceNeed, Placement } from '../../../types/child'
 import { ChildContext } from '../../../state'
-import {
-  createNewServiceNeed,
-  deleteNewServiceNeed,
-  updateNewServiceNeed
-} from '../../../api/child/new-service-needs'
-import { UIContext } from '../../../state/ui'
-
-interface DropdownOption {
-  label: string
-  value: string
-}
+import { deleteNewServiceNeed } from '../../../api/child/new-service-needs'
+import { DateRange, getGaps } from '../../../utils/date'
```

Comment:
True, but that one is a class instead of an interface so using it here gets tricky. Now service needs and gaps implement the same interface which makes rendering them in the same table easier.

Reply:
Aren't you conditionally rendering `MissingServiceNeedRow` anyway? I understand that the `DateRange` you're using here is a bit easier to use, but I think moving towards having one implementation is more important.

Reply:
Okey, I'll give it a try